Lib/test/test_pydoc/test_pydoc.py

```diff
@@ -693,6 +694,102 @@ def test_help_output_redirect(self):
         finally:
             pydoc.getpager = getpager_old
 
+    @unittest.skipIf(hasattr(sys, 'gettrace') and sys.gettrace(),
+                     'trace function introduces __locals__ unexpectedly')
+    @requires_docstrings
+    def test_help_output_redirect_various_requests(self):
+        # issue 940286, if output is set in Helper, then all output from
+        # Helper.help should be redirected
+
+        def run_pydoc_for_request(request, expected_text_part):
+            """Helper function to run pydoc with its output redirected"""
+            with captured_output('stdout') as output, captured_output('stderr') as err:
+                buf = StringIO()
+                helper = pydoc.Helper(output=buf)
+                helper.help(request)
+                result = buf.getvalue().strip()
+                self.assertEqual('', output.getvalue(), msg=f'failed on request "{request}"')
+                self.assertEqual('', err.getvalue(), msg=f'failed on request "{request}"')
+                self.assertIn(expected_text_part, result, msg=f'failed on request "{request}"')
+
+        getpager_old = pydoc.getpager
+        getpager_new = lambda: (lambda x: x)
+        self.maxDiff = None
+
+        pydoc.getpager = getpager_new
+        try:
```

Comment:
You can use `test.support.swap_attr()` or `unittest.mock.patch()`.

It would be nice to test also that the pager was not called (otherwise why do you need to patch it?).

Reply:
Good idea, done.

I modified also the existing test `test_help_output_redirect` from which I borrowed the "getpager" handling code. To me it also seems more reasonable to use mocking here.